cardscan-base/src/main/java/com/getbouncer/cardscan/base/ScanStats.java

```diff
@@ -1,21 +1,29 @@
 package com.getbouncer.cardscan.base;
 
+import android.Manifest;
+import android.app.Activity;
+import android.content.pm.PackageManager;
 import android.os.Build;
 import android.os.SystemClock;
 import android.text.TextUtils;
 
+import androidx.core.content.ContextCompat;
+
 import org.json.JSONException;
 import org.json.JSONObject;
 
 public class ScanStats {
+    private Activity parentActivity;
     private long endTimeMs;
     private long startTimeMs;
     private int scans;
     private boolean success;
+    private long panFirstDetectedAtMs = -1;
 
-    ScanStats() {
+    ScanStats(Activity activity) {
```

Comment:
instead of passing the full activity here, can we find a way to just set a boolean on scanstats if the permission was set?

Reply:
hah yes, smart